### ISSUE TITLE
macos: DefaultCW: wait_pid ChildProcessError after SIGTERM

### ISSUE THREAD

**Issue by byllyfish** [byllyfish/shellous#168]
```
tests/test_shellous.py::test_command_iterator_api_interrupted_sync 
-------------------------------- live log setup --------------------------------
1637257385.569 DEBUG asyncio Using selector: KqueueSelector
1637257385.569 DEBUG asyncio Using selector: KqueueSelector
1637257385.570 DEBUG asyncio Close <_UnixSelectorEventLoop running=False closed=False debug=True>
-------------------------------- live log call ---------------------------------
1637257385.572 DEBUG asyncio Using selector: KqueueSelector
1637257385.573 INFO shellous Runner.__aenter__ stepin <Runner 'echo' pid=None> (macOS-10.16 CPython 3.9.8 asyncio.unix_events._UnixSelectorEventLoop MainThread DefaultChildWatcher)
1637257385.573 INFO shellous Runner._start stepin <Runner 'echo' pid=None>
1637257385.573 INFO shellous Runner._subprocess_spawn stepin <Runner 'echo' pid=None>
1637257385.575 INFO shellous Runner._subprocess_exec stepin <Runner 'echo' pid=None>
1637257385.576 DEBUG asyncio execute program '/Users/runner/Library/Caches/pypoetry/virtualenvs/shellous-fzoxJzh3-py3.9/bin/python' stdin=<pipe> stdout=<pipe> stderr=<_io.TextIOWrapper name='<stderr>' mode='w' encoding='utf-8'>
1637257385.576 DEBUG shellous Thread 'KQueueAgent' starting
1637257385.580 DEBUG asyncio process '/Users/runner/Library/Caches/pypoetry/virtualenvs/shellous-fzoxJzh3-py3.9/bin/python' created: pid 11257
1637257385.585 DEBUG asyncio Write pipe 13 connected: (<_UnixWritePipeTransport fd=13 idle bufsize=0>, <WriteSubprocessPipeProto fd=0 pipe=<_UnixWritePipeTransport fd=13 idle bufsize=0>>)
1637257385.587 DEBUG asyncio Read pipe 14 connected: (<_UnixReadPipeTransport fd=14 polling>, <ReadSubprocessPipeProto fd=1 pipe=<_UnixReadPipeTransport fd=14 polling>>)
1637257385.589 INFO asyncio execute program '/Users/runner/Library/Caches/pypoetry/virtualenvs/shellous-fzoxJzh3-py3.9/bin/python': <_UnixSubprocessTransport pid=11257 running stdin=<_UnixWritePipeTransport fd=13 idle bufsize=0> stdout=<_UnixReadPipeTransport fd=14 polling>>
1637257385.589 INFO shellous Runner._subprocess_exec stepout <Runner 'echo' pid=11257 exit_code=None> ex=None
1637257385.590 INFO shellous Runner._subprocess_spawn stepout <Runner 'echo' pid=11257 exit_code=None> ex=None
1637257385.592 INFO shellous Runner._start stepout <Runner 'echo' pid=11257 exit_code=None> ex=None
1637257385.592 INFO shellous Runner.__aenter__ stepout <Runner 'echo' pid=11257 exit_code=None> ex=None
1637257385.592 INFO shellous read_lines stepin
1637257385.593 INFO shellous write_stream stepin
1637257385.594 INFO shellous write_stream stepout ex=None
1637257385.594 INFO shellous Runner._waiter stepin <Runner 'echo' pid=11257 exit_code=None>
1637257385.621 INFO shellous Runner.__aexit__ stepin <Runner 'echo' pid=11257 exit_code=None> exc_value=CancelledError()
1637257385.621 INFO shellous Runner._finish stepin <Runner 'echo' pid=11257 exit_code=None>
1637257385.621 INFO shellous Runner._kill stepin <Runner 'echo' cancelled pid=11257 exit_code=None>
1637257385.622 INFO shellous Runner.signal <Runner 'echo' cancelled pid=11257 exit_code=None> signal=<Signals.SIGTERM: 15>
1637257385.622 DEBUG shellous _reap_pid pid=11257
1637257385.622 ERROR shellous wait_pid(11257) status is 255 ex=ChildProcessError(10, 'No child processes')
1637257385.623 INFO shellous Runner._waiter stepout <Runner 'echo' cancelled pid=11257 exit_code=None> ex=CancelledError()
1637257385.624 INFO asyncio <_UnixSubprocessTransport pid=11257 running stdin=<_UnixWritePipeTransport closed fd=13 closed> stdout=<_UnixReadPipeTransport fd=14 polling>> exited with return code 255
1637257385.625 INFO asyncio <_UnixReadPipeTransport fd=14 polling> was closed by peer
1637257385.626 INFO shellous harvest_wait cancelled trustee=<Runner 'echo' cancelled pid=11257 exit_code=255> cancel_finish=False ex=GeneratorExit()
1637257385.628 INFO shellous read_lines stepout ex=GeneratorExit()
1637257385.628 WARNING shellous Runner.kill <Runner 'echo' cancelled pid=11257 exit_code=255> ex=GeneratorExit()
1637257385.628 INFO shellous Runner._kill_wait stepin <Runner 'echo' cancelled pid=11257 exit_code=255>
1637257385.628 INFO shellous Runner._kill_wait stepout <Runner 'echo' cancelled pid=11257 exit_code=255> ex=GeneratorExit()
1637257385.628 INFO shellous Runner._kill stepout <Runner 'echo' cancelled pid=11257 exit_code=255> ex=GeneratorExit()
1637257385.628 INFO shellous Runner._close stepin <Runner 'echo' cancelled pid=11257 exit_code=255>
1637257385.631 INFO shellous Runner._close stepout <Runner 'echo' cancelled pid=11257 exit_code=255> ex=GeneratorExit()
1637257385.631 INFO shellous Runner._finish stepout <Runner 'echo' cancelled pid=11257 exit_code=255> ex=GeneratorExit()
1637257385.631 INFO shellous Runner.__aexit__ stepout <Runner 'echo' cancelled pid=11257 exit_code=255> ex=GeneratorExit() exc_value=CancelledError()
1637257385.634 DEBUG asyncio Close <_UnixSelectorEventLoop running=False closed=False debug=True>
PASSED
```

**Comment by byllyfish** [comment]
```
tests/test_shellous.py::test_command_iterator_api_interrupted 
-------------------------------- live log setup --------------------------------
1637257454.994 DEBUG asyncio Using selector: KqueueSelector
1637257454.995 DEBUG asyncio Using selector: KqueueSelector
1637257454.995 DEBUG asyncio Close <_UnixSelectorEventLoop running=False closed=False debug=True>
-------------------------------- live log call ---------------------------------
1637257454.998 INFO shellous Runner.__aenter__ stepin <Runner 'echo' pid=None> (macOS-10.16 CPython 3.10.0 asyncio.unix_events._UnixSelectorEventLoop MainThread DefaultChildWatcher)
1637257454.998 INFO shellous Runner._start stepin <Runner 'echo' pid=None>
1637257454.999 INFO shellous Runner._subprocess_spawn stepin <Runner 'echo' pid=None>
1637257455.000 INFO shellous Runner._subprocess_exec stepin <Runner 'echo' pid=None>
1637257455.001 DEBUG asyncio execute program '/Users/runner/Library/Caches/pypoetry/virtualenvs/shellous-fzoxJzh3-py3.10/bin/python' stdin=<pipe> stdout=<pipe> stderr=<_io.TextIOWrapper name='<stderr>' mode='w' encoding='utf-8'>
1637257455.001 DEBUG shellous Thread 'KQueueAgent' starting
1637257455.005 DEBUG asyncio process '/Users/runner/Library/Caches/pypoetry/virtualenvs/shellous-fzoxJzh3-py3.10/bin/python' created: pid 12441
1637257455.009 DEBUG asyncio Write pipe 10 connected: (<_UnixWritePipeTransport fd=10 idle bufsize=0>, <WriteSubprocessPipeProto fd=0 pipe=<_UnixWritePipeTransport fd=10 idle bufsize=0>>)
1637257455.011 DEBUG asyncio Read pipe 11 connected: (<_UnixReadPipeTransport fd=11 polling>, <ReadSubprocessPipeProto fd=1 pipe=<_UnixReadPipeTransport fd=11 polling>>)
1637257455.012 INFO asyncio execute program '/Users/runner/Library/Caches/pypoetry/virtualenvs/shellous-fzoxJzh3-py3.10/bin/python': <_UnixSubprocessTransport pid=12441 running stdin=<_UnixWritePipeTransport fd=10 idle bufsize=0> stdout=<_UnixReadPipeTransport fd=11 polling>>
1637257455.012 INFO shellous Runner._subprocess_exec stepout <Runner 'echo' pid=12441 exit_code=None> ex=None
1637257455.013 INFO shellous Runner._subprocess_spawn stepout <Runner 'echo' pid=12441 exit_code=None> ex=None
1637257455.015 INFO shellous Runner._start stepout <Runner 'echo' pid=12441 exit_code=None> ex=None
1637257455.015 INFO shellous Runner.__aenter__ stepout <Runner 'echo' pid=12441 exit_code=None> ex=None
1637257455.015 INFO shellous read_lines stepin
1637257455.016 INFO shellous write_stream stepin
1637257455.016 INFO shellous write_stream stepout ex=None
1637257455.017 INFO shellous Runner._waiter stepin <Runner 'echo' pid=12441 exit_code=None>
1637257455.045 INFO shellous Runner.__aexit__ stepin <Runner 'echo' pid=12441 exit_code=None> exc_value=GeneratorExit()
1637257455.045 INFO shellous Runner._finish stepin <Runner 'echo' pid=12441 exit_code=None>
1637257455.045 INFO shellous Runner._kill stepin <Runner 'echo' pid=12441 exit_code=None>
1637257455.046 INFO shellous Runner.signal <Runner 'echo' pid=12441 exit_code=None> signal=<Signals.SIGTERM: 15>
1637257455.046 DEBUG shellous _reap_pid pid=12441
1637257455.047 ERROR shellous wait_pid(12441) status is 255 ex=ChildProcessError(10, 'No child processes')
1637257455.048 INFO asyncio <_UnixSubprocessTransport pid=12441 running stdin=<_UnixWritePipeTransport closed fd=10 closed> stdout=<_UnixReadPipeTransport fd=11 polling>> exited with return code 255
1637257455.049 INFO asyncio <_UnixReadPipeTransport fd=11 polling> was closed by peer
1637257455.050 INFO shellous Runner._waiter stepout <Runner 'echo' pid=12441 exit_code=255> ex=None
1637257455.052 INFO shellous read_lines stepout ex=GeneratorExit()
1637257455.052 INFO shellous Runner._kill stepout <Runner 'echo' pid=12441 exit_code=255> ex=GeneratorExit()
1637257455.052 INFO shellous Runner._close stepin <Runner 'echo' pid=12441 exit_code=255>
1637257455.055 INFO shellous Runner._close stepout <Runner 'echo' pid=12441 exit_code=255> ex=GeneratorExit()
1637257455.055 INFO shellous Runner._finish stepout <Runner 'echo' pid=12441 exit_code=255> ex=GeneratorExit()
1637257455.055 INFO shellous Runner.__aexit__ stepout <Runner 'echo' pid=12441 exit_code=255> ex=GeneratorExit() exc_value=GeneratorExit()
PASSED
```

**Comment by byllyfish** [comment]
This is normal behavior. Using waitid on the pidfd itself doesn't help. There is still an occasional ChildProcessErrror when we reap.

I made a fix in 0.10.0 to check for exit status 255 and re-map it to -SIGNAL if necessary.
